pkg/reconciler/revision/background.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revision
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/workqueue"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+// imageResolver is an interface used mostly to mock digestResolver for tests.
```

Comment:
then it should be defined in the test file?

Reply:
well, it still need to exist for it to be the type the constructor asks for?